src/test/java/com/scalar/database/storage/cassandra4driver/ClusterManagerTest.java

```diff
@@ -0,0 +1,208 @@
+package com.scalar.database.storage.cassandra4driver;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.CqlSessionBuilder;
+import com.datastax.oss.driver.api.core.DriverException;
+import com.datastax.oss.driver.api.core.metadata.Metadata;
+import com.datastax.oss.driver.api.core.metadata.schema.KeyspaceMetadata;
+import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata;
+import com.scalar.database.config.DatabaseConfig;
+import com.scalar.database.exception.storage.ConnectionException;
+import com.scalar.database.exception.storage.StorageRuntimeException;
+import java.util.Optional;
+import java.util.Properties;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/** */
+public class ClusterManagerTest {
+  private static final String VALID_ADDRESS = "127.0.0.1";
+  private static final String INVALID_PORT = Integer.toString(Integer.MAX_VALUE);
+  private static final String ANY_USERNAME = "username";
+  private static final String ANY_PASSWORD = "password";
+  private static final String ANY_KEYSPACE_NAME = "keyspace";
+  private static final String ANY_TABLE_NAME = "table";
+  @Mock private CqlSession session;
+  @Mock private CqlSessionBuilder builder;
+  private ClusterManager manager;
+  private Properties props;
+  private DatabaseConfig config;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+
+    props = new Properties();
+    props.setProperty(DatabaseConfig.CONTACT_POINTS, VALID_ADDRESS);
+    props.setProperty(DatabaseConfig.USERNAME, ANY_USERNAME);
+    props.setProperty(DatabaseConfig.PASSWORD, ANY_PASSWORD);
+    config = new DatabaseConfig(props);
+
+    manager = Mockito.spy(new ClusterManager(config));
+    doReturn(builder).when(manager).getBuilder();
+    when(builder.build()).thenReturn(session);
+  }
+
+  @Test
+  public void getSession_CalledOnce_ShouldConnectAndReturn() {
+    // Act Assert
+    assertThatCode(
+            () -> {
+              manager.getSession();
+            })
+        .doesNotThrowAnyException();
+
+    // Assert
+    verify(manager).getBuilder();
+    verify(builder).build();
+  }
+
+  @Test
+  public void getSession_ConfigWithInvalidPortGiven_ShouldThrowExceptionProperly() {
```

Comment:
I'm wondering why this test is checking an invalid port instead of an invalid endpoint.
The new driver won't really throw exception any more even if an invalid endpoint is given ?
Maybe `InetSocketAddress` won't throw ?

Reply:
`InetSocketAddress` constractor doesn't throw `IllegalArgumentException` for an invalid endpoint.
I added a check if an endpoint can be resolved and add a test for an invalid endpoint.

Reply:
Looking good !